python/ray/tune/impl/tuner_internal.py

```diff
@@ -523,7 +525,11 @@ def setup_create_experiment_checkpoint_dir(
         """Sets up experiment checkpoint dir before actually running the experiment."""
         path = Experiment.get_experiment_checkpoint_dir(
             trainable,
-            run_config.storage_path,
+            # TODO(justinvyu): This is a result of the old behavior of parsing
+            # local vs. remote storage paths and should be reworked in a follow-up.
+            _get_defaults_results_dir()
+            if _use_storage_context()
+            else run_config.storage_path,
```

Comment:
Note to self: Even though the old path takes in a `storage_path`, the `get_experiment_checkpoint_dir` method resolves the storage path and only takes the "local" version, which is the cache path in the case of cloud storage.